lib/SILOptimizer/Mandatory/MoveOnlyAddressCheckerUtils.cpp

```diff
@@ -1802,6 +1894,17 @@ bool GatherUsesVisitor::visitUse(Operand *op) {
   }
 
   if (auto *pas = dyn_cast<PartialApplyInst>(user)) {
+    if (auto *fArg = dyn_cast<SILFunctionArgument>(
+            stripAccessMarkers(markedValue->getOperand()))) {
+      // If we are processing an inout convention and we emitted an error on the
+      // partial_apply, we shouldn't process this mark_must_check, but squelch
+      // the compiler doesn't understand error.
+      if (fArg->getArgumentConvention().isInoutConvention() &&
+          pas->getCalleeFunction()->hasSemanticsAttr(
+              semantics::NO_MOVEONLY_DIAGNOSTICS)) {
```

Comment:
NOTE: This isn't actually a bug. I just by mistake left this in this specific commit. I filled in the if statement body here: https://github.com/apple/swift/pull/65871/commits/de5737df9515b67949e3c1adbb372633313bfe71#diff-9c8ec5474745801be2148dbcd9e1178e6aaead4739668d88d551b4137047cb8b

Reply:
Specifically in 052ecd1ae1930aa119e9c028dd18e7e6f118ffe3